### ISSUE TITLE
Problem with isBefore function

### ISSUE THREAD

**Issue by sh977218** [date-fns/date-fns#3369]
There is an issue with the [isBefore function](https://date-fns.org/v2.29.3/docs/isBefore)

  ## The problem
when passing a non date string, the function doesn't throw error.

  ### 💻 Code demonstrating the problem

  

  ```js
  import { isBefore } from 'date-fns'

  isBefore('None', '2022-02-02')
  //=> false
  ```

  ### 🙁 Actual behavior
isBefore returns false

  ### 🙂 Expected behavior

Throw error.

  ## Debug information

  - date-fns version: 
  - Browser/Node.js version: 
  - Your timezone: 
  - Your current time:

**Comment by rakeshbabu23** [comment]
Hii please assign me assign me this issue.I think we can solve this issue by modifying  the isBefore and isAfter  function to throw an error when it is passed a non-date string by checking the type of the input arguments using the typeof operator.

we need to modify the toDate in such a way that it should accept string because we have used string parameter in isBefore and isAfter functions.
Here are the modified codes.
isAfter
<img width="613" alt="isAFter" src="https://user-images.githubusercontent.com/120632351/233766213-75a6fe94-bd27-4007-85e8-38631b44eb2c.png">

isBefore
<img width="641" alt="isBefore" src="https://user

toDate
<img width="531" alt="toDate" src="https://user-images.githubusercontent.com/120632351/233766220-77328b24-b571-4ea7-a3ca-3ee864c75fc4.png">
-images.githubusercontent.com/120632351/233766219-5af7111c-0554-44ae-8c19-0f6744a063a2.png">

if you feel above modifications  are correct please assign me this issue.I will be happy if you suggest  any modifcations to above code

**Comment by sh977218** [comment]
I don't think this post is visible to repo's maintainer. Not sure how to get this issue escalated
